### PR TITLE
16 trail

### PR DESCRIPTION
**Pull request by IsabelleTornqvist**
Trail for message

### CODE REVIEW

File: Application/src/visuals/Message.java [hunk 1]
```diff
@@ -5,10 +5,13 @@
 import javafx.scene.image.Image;
 import visuals.handlers.Animation;
 
+import java.util.ArrayList;
+
 /**
  * Class for creating the messages to pass between "classes".
  * @author Sebastian Fransson
- * @version 1.0
+ * @version 2.0
```
Comment (ghost) [inline Application/src/visuals/Message.java:13]: version 1.1?

**Review by ghost — APPROVED** [review]

File: Application/src/visuals/Message.java [hunk 2]
```diff
@@ -20,13 +23,19 @@
...
+    private ArrayList<Trail> trails = new ArrayList<>(); //Stores the trails that appear after the dragons
 
     //Images for dragon animation.
     private static Image dragonMessage = new Image("resources/DragonBro.png"); //Wings Up
     private static Image dragonMessage2 = new Image("resources/DragonBro2.png"); //Wings Down
     private static Image dragonMessageRev = new Image("resources/DragonBroRev.png"); //Rev Wings Up
     private static Image dragonMessageRev2 = new Image("resources/DragonBroRev2.png"); //Rev Wings Down
 
+    //Image for trail animation
+    private static Image smoke = new Image("resources/cloud1.png");
```
Comment (timjon) [inline Application/src/visuals/Message.java]: Should it be called trailed or something else for clarity

File: Application/src/visuals/Message.java [hunk 3]
```diff
@@ -111,8 +122,12 @@ public void render(GraphicsContext gc){
 
         y1 += offset; // Sets an offset from the previous message.
 
+        //this.trails.add(new Trail((x1)+animationBounds, y1 +(this.class_size),
+         //       class_size/messageScale, class_size/messageScale));
+
```
Comment (timjon) [inline Application/src/visuals/Message.java]: Removed this since its commented out anyway?

File: Application/src/visuals/Message.java [hunk 4]
```diff
@@ -147,8 +170,10 @@ else if(switchImage){
                         y1 + (this.class_size), class_size/messageScale, class_size/messageScale); //State Wings Down.
                 switchImage = true;
             }
-
         }
+        //Draw trail for message, for each instance in the arraylist
```
Comment (timjon) [inline Application/src/visuals/Message.java]: for THE message?

File: Application/src/visuals/Trail.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    //Getter methods for coordinates, height and width
+    public int getXcoordinate(){return  xcoordinate;}
+
+    public int getYcoordinate(){return  ycoordinate;}
+
+    public double getWidth(){return width;}
+
+    public double getHeight(){return height;}
```
Comment (timjon) [inline Application/src/visuals/Trail.java]: Missing some comments for the methods here

**Review by timjon — CHANGES_REQUESTED** [review]

File: Application/src/visuals/Message.java (hunk 1 above)
Comment (IsabelleTornqvist) [reply Application/src/visuals/Message.java:13]: new feature makes it 2.0

**Review by timjon — CHANGES_REQUESTED** [review]
The trail does not scale with the application
And do we want javadoc comments for methods?

File: Application/src/visuals/Trail.java [hunk 6]
```diff
@@ -23,16 +23,28 @@ public Trail(int xcoordinate, int ycoordinate, double width, double height){
...
-    //Get method for getting the width of the trail image
+    /**
+     * Getter method for getting the width of the trail image
+     * @return width
+     */
     public double getWidth(){return width;}
 
-    //Get method for getting the height of the trail image
+    /**
+     * Getter method for getting the height od the trail image
```
Comment (ghost) [inline Application/src/visuals/Trail.java]: of*

File: Application/src/visuals/Trail.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
+package visuals;
+
+/**
+ * Class for the trail of the message
```
Comment (pontuslaestadius) [inline Application/src/visuals/Trail.java]: Don't make the header dependable on another class. Just say what this class does, not what it's currently implemented in.Basically make it more generic, and remove mention of message.

**Review by pontuslaestadius — APPROVED** [review]
Added scaling on defect backlog

**Review by timjon — APPROVED** [review]

**Pull request merged**